exec/csnex/Exec.cs

```diff
@@ -813,7 +813,26 @@ void INDEXAV()
 
         void INDEXAN()
         {
-            throw new NotImplementedException(string.Format("{0} not implemented.", MethodBase.GetCurrentMethod().Name));
+            ip++;
+            Number index = stack.Pop().Number;
+            Cell array = stack.Pop();
+
+            if (!index.IsInteger()) {
+                RaiseLiteral("ArrayIndexException", Cell.CreateStringCell(index.ToString()));
+                return;
+            }
+            int i = Number.number_to_int32(index);
+            if (i < 0) {
+                RaiseLiteral("ArrayIndexException", Cell.CreateStringCell(new Number(i).ToString()));
+                return;
+            }
+            if (i >= array.Array.Count) {
```

Comment:
Probably better to combine these two tests (this and the one above) into an 'or' condition, since the exception raised is identical.